Formula/das_tool.rb

```diff
@@ -0,0 +1,42 @@
+class DasTool < Formula
+  # cite Sieber_2018: "https://doi.org/10.1038/s41564-018-0171-1"
+  desc "Genomic binning refiner"
+  homepage "https://github.com/cmks/DAS_Tool"
+  url "https://github.com/cmks/DAS_Tool/archive/1.1.1.tar.gz"
+  sha256 "2a55f67b5331251d8fd5adea867cc341363fbf7fa7ed5c3ce9c7679d8039f03a"
+  head "https://github.com/cmks/DAS_Tool.git"
+
+  if OS.mac?
+    depends_on "coreutils"
+    depends_on "gettext"
+  else
+    depends_on "icu4c"
+    depends_on "unzip" => :build
+  end
+
+  depends_on "diamond"
+  depends_on "prodigal"
+  depends_on "pullseq"
+  depends_on "r"
```

Comment:
On macOS, it's much better to use the cask `r-app` rather than the brewed `r`, because the former can use precompiled binary packages from CRAN, but the latter cannot, due to its installation directory.
Let's make this a `:build` dependency on macOS. You could add a `def caveats` that instructs the user to `brew cask install r-app` on macOS. Not an ideal situation I know.

Reply:
Assuming `brew test das_tool` passes without R installed. There's a workaround if it fails. You can add dependencies for testing only.

Reply:
The test is somewhat of a pseudo-test, so it should pass. I'm wondering then you if you would prefer the user to add the packages manually, rather than install them in the package directory. Will the bottle linkages work if `r-app` is installed, rather than brew `r`?

```
❯ brew linkage --reverse das_tool
...
/usr/local/opt/r/lib/R/lib/libR.dylib
  libexec/lib/R/Rcpp/libs/Rcpp.so
  libexec/lib/R/colorspace/libs/colorspace.so
  libexec/lib/R/data.table/libs/datatable.so
  libexec/lib/R/digest/libs/digest.so
  libexec/lib/R/fansi/libs/fansi.so
  libexec/lib/R/glue/libs/glue.so
  libexec/lib/R/lazyeval/libs/lazyeval.so
  libexec/lib/R/plyr/libs/plyr.so
  libexec/lib/R/reshape2/libs/reshape2.so
  libexec/lib/R/rlang/libs/rlang.so
  libexec/lib/R/scales/libs/scales.so
  libexec/lib/R/stringi/libs/stringi.so
  libexec/lib/R/tibble/libs/tibble.so
  libexec/lib/R/utf8/libs/utf8.so
```

Reply:
I pushed the change, though I'm unsure if it'll work. Perhaps best to write a proper test. There's example data, though it's not a simple test.

Reply:
> I'm wondering then you if you would prefer the user to add the packages manually, rather than install them in the package directory. Will the bottle linkages work if r-app is installed, rather than brew r?

Hmm. Right. If there were no native (compiled) packages, it'd work just fine. Since there are native packages, you're right, it probably won't work as is.

The two options are to use the brewed R on both Linux and macOS and install the packages as you've done on both. Alternatively, we could install the packages only for Linux, and on macOS display a caveat instructing the user to `brew cask install r-app` and `Rscript -e "install.packages(…)"`. Which do you prefer?

Reply:
I have a slight preference for the latter. I generally think it's a mistake to install the brewed R on macOS, since the cask works better, and having both installed causes confusion.

Reply:
I'm fine with either. Personally I've been using the brewed R for a while, so I would have leaned towards that. Maybe the very best would be to put a runtime check for the packages in a wrapper script that notifies the user if they are not installed?

Reply:
Something like this?

```
!/bin/bash
Rscript -e 'library(ggplot2);library(data.table);library(doMC);' 2&>1 /dev/null
if [ $? -ne 0 ]; then
  echo "DAS_Tool requires the packages 'ggplot2', 'data.table', "
  echo "and 'doMC' to be installed. Please install them before use "
  echo "with "
  echo ""
  echo "  Rscript -e 'install.packages(c(\"ggplot2\",\"data.table\",\"doMC\"))'"
  echo ""
else
```

Reply:
Sure that works for me. For style points, I'd suggest using `cat <<EOS` rather than `echo`.

Reply:
Actually, I just realised that the package won't install unless R is already installed:

```
 system "R", "CMD", "INSTALL", "./package/DASTool_1.1.1.tar.gz"
```

Is there a way to require either 'r-app' or 'r' be installed a priori?

Reply:
Let's use the brewed R on both macOS and Linux. Sounds simpler. Sorry for the runaround.